pkg/sync/v6/client.go

```diff
@@ -233,6 +233,12 @@ func needsUpdate(log *logrus.Entry, existing, o *unstructured.Unstructured) bool
 		return false
 	}
 
+	// check if object is marked deploy-only and deploy once
+	if o.GetAnnotations()[deployOnceSyncPodLabelKey] != "" {
```

Comment:
can you have a think now about how this mechanism could be improved to in the future support objects that we might allow the end user to delete - e.g. templates in the openshift namespace?  If we can do that now, it might be useful.

Reply:
Azure Arch document already has topic for this.